pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -252,6 +252,104 @@ pub(super) async fn download_index_part(
     Ok(index_part)
 }
 
+pub(super) async fn download_index_part(
+    storage: &GenericRemoteStorage,
+    tenant_id: &TenantId,
+    timeline_id: &TimelineId,
+    my_generation: Generation,
+) -> Result<IndexPart, DownloadError> {
+    if my_generation.is_none() {
+        // Operating without generations: just fetch the generation-less path
+        return do_download_index_part(storage, tenant_id, timeline_id, my_generation).await;
+    }
+
+    let previous_gen = my_generation.previous();
+    let r_previous = do_download_index_part(storage, tenant_id, timeline_id, previous_gen).await;
+
+    match r_previous {
+        Ok(index_part) => {
+            tracing::debug!("Found index_part from previous generation {previous_gen:?}");
```

Comment:
please add a brief reference to RFC's section about falling back why this shortcut is correct.

Reply:
Added comments in c1ca7737a099f3f657626218145de8dacff8bb2d